modules/ndu/src/backend/training/model-service.ts

```diff
@@ -0,0 +1,92 @@
+import * as sdk from 'botpress/sdk'
+import crypto from 'crypto'
+import fse, { WriteStream } from 'fs-extra'
+import _ from 'lodash'
+import path from 'path'
+import { Stream } from 'stream'
+import tar from 'tar'
+import tmp from 'tmp'
+
+export interface Model {
+  hash: string
+  startedAt: Date
+  finishedAt: Date
+  success: boolean
+  data: any
+}
+
+export const MODELS_DIR = './models'
+const MAX_MODELS_TO_KEEP = 2
+
+function makeFileName(hash: string): string {
+  return `${hash}.ndu.model`
+}
+
+export async function pruneModels(ghost: sdk.ScopedGhostService): Promise<void | void[]> {
+  const models = await listModels(ghost)
+  if (models.length > MAX_MODELS_TO_KEEP) {
+    return Promise.map(models.slice(MAX_MODELS_TO_KEEP), file => ghost.deleteFile(MODELS_DIR, file))
+  }
+}
+
+export async function listModels(ghost: sdk.ScopedGhostService): Promise<string[]> {
+  return ghost.directoryListing(MODELS_DIR, '*', undefined, undefined, {
+    sortOrder: { column: 'modifiedOn', desc: true }
+  })
+}
+
+export async function getModel(ghost: sdk.ScopedGhostService, hash: string): Promise<Model | undefined> {
+  const fileName = makeFileName(hash)
+  if (!(await ghost.fileExists(MODELS_DIR, fileName))) {
+    return
+  }
+  const buffStream = new Stream.PassThrough()
+  buffStream.end(await ghost.readFileAsBuffer(MODELS_DIR, fileName))
+  const tmpDir = tmp.dirSync({ unsafeCleanup: true })
+
+  const tarStream = tar.x({ cwd: tmpDir.name, strict: true }, ['model']) as WriteStream
```

Comment:
Not sure its necessary to compress ndu models as they're quite small.

Reply:
I copied the logic from NLU, in case we end up making a service to handle models (eg, if we end up using the largeObject API). And if we ever end up with larger models, it will already be taken care of